### PR TITLE
build: precheck the length of CHART_VERSION

### PR DESCRIPTION
**Pull request by Vicente-Cheng**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

**Problem:**
If we create a long branch name, we will encounter the following error when we install ISO from it.
```
harvester-node-0:~ # kubectl get bundle -A
NAMESPACE     NAME                                          BUNDLEDEPLOYMENTS-READY   STATUS
fleet-local   fleet-agent-local                             1/1
fleet-local   local-managed-system-agent                    1/1
fleet-local   mcc-harvester                                 0/1                       ErrApplied(1) [Cluster fleet-local/local: failed to create resource: ServiceAccount "harvester" is invalid: metadata.labels: Invalid value: "harvester-0.0.0-add-new-cluster-role-for-csi-driver-2d8c77fe-dirty": must be no more than 63 characters]
fleet-local   mcc-harvester-crd                             1/1
fleet-local   mcc-local-managed-system-upgrade-controller   1/1
fleet-local   mcc-rancher-logging-crd                       1/1
fleet-local   mcc-rancher-monitoring-crd                    1/1
```

**Solution:**
We can do some precheck to avoid it.

**Related Issue:**
https://github.com/harvester/harvester/issues/4882

**Test plan:**
1. Create a long enough branch name to build.
2. It should fail.

### CODE REVIEW

File: scripts/version [hunk 1]
```diff
@@ -38,6 +38,15 @@ fi
 # Drop the v prefix for Chart Version to follow existing pattern.
 CHART_VERSION="$(echo ${CHART_VERSION} | sed -E 's/^v//')"
 
+# The length of metadata.label is no more than 63 characters.
+# We limit the CHART_VERSION lees than 50 chars and it would be safe.
```
Comment (starbops) [inline scripts/version]: typo here
```suggestion
# We limit the CHART_VERSION less than 50 chars and it would be safe.
```

Comment (starbops) [reply scripts/version]: maybe change it to "less than or equal to 50 chars" to be more precise?

File: scripts/version [hunk 2]
```diff
@@ -38,6 +38,15 @@ fi
...
+CHART_VERSION_LEN=${#CHART_VERSION}
+if [[ $CHART_VERSION_LEN > 50 ]]; then
+    echo "Please reduce the length of CHART_VERSION, it should less than 50."
```
Comment (starbops) [inline scripts/version]: ditto

**Review by starbops — COMMENTED** [review]
The functionality LGTM. Thanks for the enhancement!

---

Check out to a new branch with a long branch name:

```
$ echo -n 'whimsical-jamboree-unfolds-with-laughter-and-joy' | wc -c
48
$ git checkout -b whimsical-jamboree-unfolds-with-laughter-and-joy
```

Build the code, and it will failed due to the check newly implemented:

```
$ make build
...
 => => exporting layers                                                                                                                                                                                 3.2s
 => => writing image sha256:3cb89e845111ad2ae32c33fa4e74a11cee2a26801e9f8e858f93acca4b1a120c                                                                                                            0.0s
 => => naming to docker.io/library/harvester:whimsical-jamboree-unfolds-with-laughter-and-joy                                                                                                           0.0s
Please reduce the length of CHART_VERSION, it should less than 50.
Current CHART_VERSION(63): 0.0.0-whimsical-jamboree-unfolds-with-laughter-and-joy-4f46fe70
FATA[0113] exit status 1                                
make: *** [build] Error 1
```

To make `CHART_VERSION` exact 50 characters long, checkout to another new branch with a shortened branch name:

```
$ echo -n 'whimsical-jamboree-unfolds-with-lau' | wc -c
35
$ git checkout -b whimsical-jamboree-unfolds-with-lau
```

This time the build is successful.

File: scripts/version (hunk 1 above)
Comment (Vicente-Cheng) [reply scripts/version]: Yap, that would be good! updated!

**Review by starbops — APPROVED** [review]

**Review by bk201 — APPROVED** [review]

**Pull request merged**